cmd/snap/cmd_run.go

```diff
@@ -216,6 +220,72 @@ func snapRunHook(snapName, snapRevision, hookName string) error {
 	return runSnapConfine(info, hook.SecurityTag(), snapName, "", hook.Name, nil)
 }
 
+func migrateXauthority(info *snap.Info) error {
+	u, err := userCurrent()
+	if err != nil {
+		return fmt.Errorf(i18n.G("cannot get the current user: %v"), err)
+	}
+
+	// If we're running as root then we don't do anything
```

Comment:
Idea was that it doesn't make sense to allow root using X11 but there could be edge cases where people do this. Regular case will be always that a user != root has an XAUTHORITY set. Will drop this. @jdstrand please correct me if I miss anything.